pkg/plugin/client/client.go

```diff
@@ -38,10 +42,42 @@ type CustomCommandExecutor interface {
 	ExecuteCustomCommand(cmdName string, ctx context.Context, args []string) error
 }
 
-type clientFactory struct{}
+type clientFactory struct {
+	bzl bazel.Bazel
+}
+
+// buildPlugin asks bazel to build the target and returns the path to the resulting binary
+func (c *clientFactory) buildPlugin(target string) (string, error) {
+	agc, err := c.bzl.AQuery(target)
+	if err != nil {
+		return "", err
+	}
+	outs := bazel.ParseOutputs(agc)
+
+	var pluginPath string
+	for _, a := range outs {
+		// TODO: don't hard-code GoLink, plugins could be written in other languages
+		if a.Mnemonic == "GoLink" {
+			pluginPath = a.Path
+			break
+		}
+	}
+	if pluginPath == "" {
+		return "", fmt.Errorf("no output file from a GoLink action was found for target %v", target)
+	}
+
+	// TODO: we should be careful to use the right flags for this build
```

Comment:
I don't think there's an actual `TODO` here. It should be a `WARNING` instead to never add flags here.